app/forms/school_search_form.rb

```diff
@@ -6,10 +6,12 @@ class SchoolSearchForm
   attr_accessor :school_name, :location, :search_distance, :search_distance_unit, :characteristics, :partnership
 
   def find_schools(page)
-    School.where(
-      "lower(name) like ?", "%#{(school_name || '').downcase}%"
-    ).includes(
-      :network, :lead_provider
-    ).page(page)
+    if partnership&.include? "partnered_with_another_provider"
+      School.where(id: Partnership.select(:school_id).map(&:school_id)).page(page)
```

Comment:
I'm not sure this is the behaviour we want. I think at a minimum this should work in conjunction with the other filters. I'm pretty sure Dom's design was that partnered schools would be excluded from the results, and only be included when this box was checked. I don't think that's what is happening here, right?

Reply:
At the minute that's how it's working, so you either get a search result from the name or by this filter. 

Definitely agree it needs to work in conjunction other filters, especially location and characteristics. I think whilst we only have this searching by school or urn I hadn't thought about working alongside just that. Happy to do that if it's worth doing from the start?

Reply:
I think so, especially as it should just be a one-liner